### PR TITLE
Bump Rust toolchain to 1.67.0 and run cargo update

### PR DESCRIPTION
**Pull request by imikushin**
To keep things fresh :)

### CODE REVIEW

**Comment by imikushin** [comment]
Fixing a LOT of clippy errors.

**Comment by imikushin** [comment]
Resolved merge conflicts.

**Comment by mimoo** [comment]
can you `git merge master` in this branch :o ?

**Comment by imikushin** [comment]
@mimoo done!

**Comment by mimoo** [comment]
sorry again, admitting the Mina side works (https://github.com/MinaProtocol/mina/pull/12613) can you update this PR again to include the Rust update in all the `CHANGELOG.md` files? (we just added changelogs to keep track of all the changes :D)

File: kimchi/src/circuits/expr.rs [hunk 1]
```diff
@@ -983,11 +983,7 @@ impl<'a, F: FftField> EvalResult<'a, F> {
         }
     }
 
-    fn add<'b, 'c>(
-        self,
-        other: EvalResult<'b, F>,
-        res_domain: (Domain, D<F>),
-    ) -> EvalResult<'c, F> {
+    fn add<'c>(self, other: EvalResult<'_, F>, res_domain: (Domain, D<F>)) -> EvalResult<'c, F> {
```
Comment (mimoo) [inline kimchi/src/circuits/expr.rs:986]: curious what the lint was there

**Review by mimoo — APPROVED** [review]
this is awesome :) thanks for going through this!

**Comment by mimoo** [comment]
merging now as things are passing! I'll create a separate PR for the changelogs

Screenshot 2023-02-08 at 9 17 01 AM

**Comment by mimoo** [comment]
merging now as things are passing! I'll create a separate PR for the changelogs

Screenshot 2023-02-08 at 9 17 01 AM

File: kimchi/src/circuits/expr.rs (hunk 1 above)
Comment (imikushin) [reply kimchi/src/circuits/expr.rs:986]: Lifetime elision

**Pull request merged**